### PR TITLE
Added kwarg: 'raise_missing_property' to 'NodePopulation.get'

### LINKED ISSUE #204: regex defined node set not working in NodePopulations that don't have the property used
Situation:
A node set is defined using a regex:
```
"mc2_Column": {
    "region": {
      "$regex": "mc2.*"
    }
  }
```
The user uses the node set with a NodePopulation:
`circ.nodes["some_population"].get("mc2_Column")`
where "some_population" does _not_ have the property "region".

Expected behavior:
An empty DataFrame is returned.

Actual behavior: 
`BluepySnapError: Unknown node properties: {'region'}`

I think it is a legitimate use case to evaluate a node set against a specific NodePopulation. As such, the user should be able to expect a result and not an error.

### PR DESCRIPTION
**Pull request by joni-herttuainen**
Fixes #204

### CODE REVIEW

File: tests/test_nodes/test_node_population.py [hunk 1]
```diff
@@ -332,6 +332,13 @@ def test_get(self):
         assert _call().shape == (3, 12)
         assert _call(0, Cell.MTYPE) == "L2_X"
         assert _call(CircuitNodeId("default", 0), Cell.MTYPE) == "L2_X"
+
+        # Unknown property filters
+        assert _call({"no-such-property": "no-value"}, raise_missing_property=False).empty
+        with pytest.raises(BluepySnapError) as e:
+            _call({"no-such-property": "no-value"})
+        assert "Unknown node properties" in e.value.args[0]
```
[the remark this answers is not in the document]
Comment (joni-herttuainen) [reply tests/test_nodes/test_node_population.py:340]: I tend to agree. Only reason I went with `assert` was that it was used in all of the other cases and I sacrificed the practicality for uniformity.

I think we should convert all the cases to use `match` instead.

File: bluepysnap/nodes/node_population.py [hunk 2]
```diff
@@ -372,14 +372,16 @@ def ids(self, group=None, limit=None, sample=None, raise_missing_property=True):
...
+    def get(self, group=None, properties=None, raise_missing_property=True):
         """Node properties as a pandas Series or DataFrame.
 
         Args:
             group (int/CircuitNodeId/CircuitNodeIds/sequence/str/mapping/None):
                 see :ref:`Group Concept`
             properties (list|str|None): If specified, return only the properties in the list.
                 Otherwise return all properties.
+            raise_missing_property (bool): if True, raises if a property used for filtering ids
+                is not listed in this population. Otherwise, no ids are selected.
```
[the remark this answers is not in the document]
Comment (joni-herttuainen) [reply bluepysnap/nodes/node_population.py:384]: yeah, I lazily copy pasted that from the `ids` with minor rewording. Thanks.

File: tests/test_nodes/test_node_population.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (GianlucaFicarelli) [reply tests/test_nodes/test_node_population.py:340]: Sure, we can convert them in a separate PR

**Comment by mgeplf** [comment]
Before this goes in, we should test that it's compatible w/ using libsonata to make node_set queries.

**Comment by joni-herttuainen** [comment]
I'm just about done with letting libsonata handle the node sets. So I'll take care of this next, then.

**Comment by joni-herttuainen** [comment]
`libsonata` raises for unknown node properties by default.

I tried to keep this in mind when doing the libsonata node set resolution, so it should be just a matter of adding and passing the argument in https://github.com/BlueBrain/snap/blob/libsonata_nodeset/bluepysnap/node_sets.py#L45, and then handling it according to `raise_missing_property`.

**Review by GianlucaFicarelli — APPROVED** [review]

**Pull request merged**